sklearn/tree/tree.py

```diff
@@ -727,6 +727,73 @@ def predict_log_proba(self, X):
 
             return proba
 
+    def predict_uplift(self, X, check_input=True):
+        """Predict uplift for X.
+        Predict probabilities for positive response for treated
+        and control groups.
+
+        Uplift probabilities are computed as
+        p[LIFT_TREAT_NOTACTIVE]   = P(0|T)
+        p[LIFT_TREAT_ACTIVE]      = P(1|T)
+        p[LIFT_CONTROL_NOTACTIVE] = P(0|C)
+        p[LIFT_CONTROL_ACTIVE]    = P(1|C)
+
+        #LIFT_TREAT_NOTACTIVE = 0
+        #LIFT_TREAT_ACTIVE = 1
+        #LIFT_CONTROL_NOTACTIVE = 2
+        #LIFT_CONTROL_ACTIVE = 3
+
+        are the indices of the returned probabilities.
+
+        Parameters
+
+
+        ----------
+        X : array-like or sparse matrix of shape = [n_samples, n_features]
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+        check_input : boolean, (default=True)
+            Allow to bypass several input checking.
+            Don't use this parameter unless you know what you do.
+        Returns
+        -------
+        p : array of shape = [n_samples, n_classes], or a list of n_outputs
+            such arrays if n_outputs > 1.
+            The uplift probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute `classes_`.
+        """
+        X = self._validate_X_predict(X, check_input)
+        proba = self.tree_.predict(X)
+
+        if self.n_outputs_ == 1:
```

Comment:
You should put an assert here to check that n_outputs == 4 I think.  It should always be 4 in the uplift scenario.